lib/PuppeteerSharp/ChromiumProcess.cs

```diff
@@ -193,7 +193,7 @@ public async Task<bool> WaitForExitAsync(TimeSpan? timeout)
                 await _exitCompletionSource.Task.WithTimeout(() =>
                 {
                     taskCompleted = false;
-                }, timeout.Value.Milliseconds).ConfigureAwait(false);
+                }, ToTimeoutInt32(timeout.Value)).ConfigureAwait(false);
```

Comment:
Why do we need this?

Reply:
`timeout.Value.Milliseconds` only returns the milliseconds portion of the TimeSpan. Instead TotalMilliseconds should be used, but that may result in a value greater than `int.MaxValue`.

Reply:
Maybe we can improve `WithTimeout`, adding an overload with a `TimeSpan` parameter (which would be the normal flow internally)

Reply:
`TimeSpan.Milliseconds` returns only the milliseconds portion of the `TimeSpan`. Instead, `TotalMilliseconds` should be used, but that returns a `double` that mat be greater than `int.MaxValue`.